### ISSUE TITLE
🐛 BUG: fsonathan... Vite error when using `fs` in external files

### ISSUE THREAD

**Issue by aFuzzyBear** [withastro/astro#2389]
### What version of `astro` are you using?

latest

### What package manager are you using?

yarn

### What operating system are you using?

wsl2; ubuntu 20.04, reproduced on stackblitz

### Describe the Bug

Hi team, 

I would like to report an issue raised in the support thread of Discord: https://discord.com/channels/830184174198718474/931342087725645834

Here I reported a `vite` error where when using nodes `fs` package in external `*.js` files, Vite throws an error, saying it cannot resolve dependencies,

This occurs with the following import statements
```js
import fs from 'fs'
import fs from 'fs/promises'
import fs from 'node:fs'
import fs from 'node:fs/promises'
```

The present work around is to disable the import statement when launching the dev server and enabling them shortly after launch, it does throw in build also prohibiting the final build since Its one wee file using `node.fs.writeFile()` 

Thanks to Muhymin for producing a reproduction,

Thank you guys for looking into this, 

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-xs2jbp?file=src%2Fpages%2Findex.astro&on=stackblitz

**Comment by jonathantneal** [comment]
Verified. If this is an intended limitation from Vite or Astro, we should document this.

**Comment by zadeviggers** [comment]
I ran into this while trying to do some stuff with `@astropub/codecs`, if that's useful to anyone working on this.

**Comment by FredKSchott** [comment]
I also just ran into this, in a file that was only ever run on the server and never meant for the client bundle. As @aFuzzyBear said, the current solution is to remove the import on dev server start and then add back after start, which feels weiiiiiiiird.

**Comment by FredKSchott** [comment]
Ah, I believe that this is a Vite bug. fixed in https://github.com/vitejs/vite/commit/4a05a6e2f69c3a799fee651d182beeb176abfaac but not yet in Astro.

**Comment by zadeviggers** [comment]
Any ideas on when the fix might land in Astro?

**Comment by aFuzzyBear** [comment]
Thanks Fred, for a while there I thought I was losing the plot there 😅

**Comment by FredKSchott** [comment]
I had the same problem! @zadeviggers asap, you can track progress in #2037

**Comment by tony-sull** [comment]
Fixed with the with the Vite upgrade in [#2424](https://github.com/withastro/astro/pull/2424) 🥳

**Comment by aFuzzyBear** [comment]
@tony-sull  legend thank you...
